### ISSUE TITLE
Idempotency breaks on Jboss::Internal::Service/Service[..] when running inside Docker container

### ISSUE THREAD

**Issue by cardil** [coi-gov-pl/puppet-jboss#47]
When running inside Docker container idempotency breaks on `Jboss::Internal::Service/Service[..]`. Puppet tries to set enable property from 'false' to 'true'.

```
[root@wfly jboss]# puppet apply tests/domain/controller.pp
notice: /Stage[main]/Jboss::Internal::Service/Service[wildfly]/enable: enable changed 'false' to 'true'
notice: /Stage[main]/Jboss::Internal::Service/Exec[jboss::service::test-running]: Triggered 'refresh' from 1 events
notice: Finished catalog run in 2.37 seconds
[root@wfly jboss]# puppet apply tests/domain/controller.pp
notice: /Stage[main]/Jboss::Internal::Service/Service[wildfly]/enable: enable changed 'false' to 'true'
notice: /Stage[main]/Jboss::Internal::Service/Exec[jboss::service::test-running]: Triggered 'refresh' from 1 events
notice: Finished catalog run in 2.50 seconds
```

This should be addressed, but it's not priority in my opinion. I think like that because use case for this is rather vague: "Managing JBoss inside a Docker container using Puppet". It's not impossible, though.

**Comment by cardil** [comment]
To @coi-gov-pl/jboss-puppet-team 

If we like to continue support Puppet in versions before 3.x (I think is a must for us), we should do something with this. Puppet improperly handles a service type's enable parameter while running in Docker. This is due to fact that `chkconfig` behaves differently in Docker and in real virtual machine and Puppet use it on Red Hat family to check is service enabled.

```
# docker
[root@ebc40bcbe0e0 /]# /sbin/chkconfig postgresql
[root@ebc40bcbe0e0 /]# echo $?
1
```

```
# VM
[root@localhost ~]# /sbin/chkconfig postgresql
[root@localhost ~]# echo $?
0
```

Solution to this is not to set enable parameter when running in Docker. Unfortunately Facter has added support for docker in version: 2.1.0. Change log here: https://github.com/puppetlabs/facter/compare/2.0.2...2.1.0

Proof:

```
[root@ebc40bcbe0e0 /]# gem install facter --version '~> 2.0.0'
Successfully installed facter-2.0.2
1 gem installed
Installing ri documentation for facter-2.0.2...
Installing RDoc documentation for facter-2.0.2...
[root@ebc40bcbe0e0 /]# facter virtual
physical
[root@ebc40bcbe0e0 /]# gem uninstall facter
Removing facter
Successfully uninstalled facter-2.0.2
[root@ebc40bcbe0e0 /]# gem install facter --version '~> 2.1.0'
Successfully installed facter-2.1.0
1 gem installed
Installing ri documentation for facter-2.1.0...
Installing RDoc documentation for facter-2.1.0...
[root@ebc40bcbe0e0 /]# facter virtual
docker
```

My proposition is to create a `jboss_deckerized` fact and to use it instead of standard `virtual` fact until dropping support for Puppet < 3.x

**Comment by cardil** [comment]
This has been fixed by PR #67
